go/arrow/flight/flightsql/driver/config.go

```diff
@@ -30,14 +74,15 @@ type DriverConfig struct {
 	Timeout  time.Duration
 	Params   map[string]string
 
-	TLSEnabled bool
-	TLSConfig  *tls.Config
+	TLSEnabled    bool
+	TLSConfigName string
+	TLSConfig     *tls.Config
 }
 
 func NewDriverConfigFromDSN(dsn string) (*DriverConfig, error) {
 	u, err := url.Parse(dsn)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("invalid URL: %w", err)
 	}
 
 	// Sanity checks on the given connection string
```

Comment:
as I'm looking at this, is there any reason that this doesn't follow the general consistency of other implementations by using a scheme of `grpc+tcp` vs `grpc+unix` vs `grpc+tls`?

Reply:
I guess that's because I wanted to keep the same scheme other SQL drivers use (see https://github.com/xo/dburl#example-urls). I think we could also support the ones above even though I don't like the redundant `grpc`...

Reply:
I think this is unrelated to this PR. Are you OK with opening an issue for this request and proceed with this PR?

Reply:
sure thing

Reply:
Created issue #35080